sdk/trace/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -286,6 +286,9 @@ public ReadWriteSpan addEvent(String name, long timestamp, TimeUnit unit) {
     if (name == null) {
       return this;
     }
+    if (unit == null) {
+      unit = TimeUnit.MILLISECONDS;
```

Comment:
That seems questionable. In that case, maybe it would be better to fall back to using the current time as if `addEvent(name)` -- and log something.

Reply:
Yeah that's true - I remembered our convention of env variables all being millisecond so thought of this. But ignoring the timestamp does seem better.

Reply:
I'm not sure what the correct behavior might be. Can we add some javadocs to the API to say that malformed inputs like this will have indeterminate results?

Reply:
Hmm - I wonder what a good technique is. One option is add a copy-paste blob to each of the `package-info` in the API artifact, another is to add it to `OpenTelemetry`, our "entry point". Or even both. Any thoughts?

Reply:
I don't have any good thoughts on how to document this in a way that doesn't mean going to each method and examining it closely for possible weird corner cases. We could go so far as to just mark the `TimeUnit` nullable and specify the behavior exactly in the API, but that also feels weird. We really don't want people to send in null parameters, but we can't stop them, so what do we do?

Reply:
I realized that there isn't so much we can do - we can expect the percentage that pass null here to not read docs to be large :) I think we just pick a behavior and stick with it - the goal is to not crash the app, it isn't to ensure telemetry works when used incorrectly.

So I don't think we should fall back to current time - it obfuscates the bug even more since the data will be slightly realistic. Instead passing a number and seeing a ridiculous timeline in the backend (more likely backends will filter them out if sophisticated enough) could make it easier to find the bug. Naturally, we do need to implement the named logger for parameter validation to present more info too.

So I would just pick a fallback unit and go with it everywhere there's a chance of an app crash. I lean towards milliseconds since we do use it for unitless timestamps elsewhere. And it's not important to have any reason since we just need something to prevent app crash.

Reply:
Another option would be to discard the event completely, like we do with null attribute values.

Or you could add an extra event attribute with error message.

Reply:
While I appreciate the desire to not crash a user's application, and I don't have all the history of previous discussions/decisions on similar topics, but is there a desire to "fail" as early as possible?

If a `null` value leads to milliseconds being used, it's highly likely this only gets noticed in monitoring systems, and even then only if the value associated with the milliseconds is outside a "normal" range of expected values. I could see the values being "normal" and therefore no problems are flagged, or that they're so bad alerts and other monitoring escalations are triggered.

Would it be reasonable to assume some level of application testing is performed before deployment to production?

Could we do the same as a `null` `name` and do nothing?

Reply:
I the risk I see is that someone writes poor library instrumentation, built in to a library, then someone's app crashes when they upgrade the library. Is that ok?

Reply:
> Or you could add an extra event attribute with error message.

@Oberon00 This is actually one of the reasons I was leaning towards not discarding, in the future adding attributes for such error messages may or may not make sense.

> While I appreciate the desire to not crash a user's application, and I don't have all the history of previous discussions/decisions on similar topics, but is there a desire to "fail" as early as possible?

There is a desire to fail at startup, but we don't want to cause anymore crashes when the user's app is actually running. These tend to be the scariest issues for users, and the reality is many users expect telemetry to get some data "for free" since it doesn't have direct business value. They're angry when the release of a new feature has to get rolled back due to crashes, but can live with some lost tracing until the next release. The spec describes it here

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md#error-handling-in-opentelemetry

and I guess it ended up like that because I'm probably not the only one that's dealt with a customer angry about crashes from a tracing SDK :)

Reply:
Discard won so went with it!